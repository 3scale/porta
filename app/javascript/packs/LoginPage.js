import React from 'react'
import { render } from 'react-dom'
import SimpleLoginPage from 'SimpleLoginPage/index'

document.addEventListener('DOMContentLoaded', () => {
  render(<SimpleLoginPage />, document.getElementById('login-page-wrapper'))
})
