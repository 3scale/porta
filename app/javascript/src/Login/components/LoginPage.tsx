import { LoginMainFooterBandItem, LoginPage as PF4LoginPage } from '@patternfly/react-core'

import { createReactWrapper } from 'utilities/createReactWrapper'
import { AuthenticationProviders } from 'Login/components/AuthenticationProviders'
import { LoginForm } from 'Login/components/LoginForm'
import brandImg from 'Login/assets/images/3scale_Logo_Reverse.png'
import PF4DownstreamBG from 'Login/assets/images/PF4DownstreamBG.svg'

import type { FunctionComponent } from 'react'
import type { ProvidersProps } from 'Login/components/AuthenticationProviders'
import type { FlashMessage } from 'Types'

interface Props {
  authenticationProviders: ProvidersProps[];
  flashMessages: FlashMessage[];
  providerSessionsPath: string;
  providerRequestPasswordResetPath: string;
  show3scaleLoginForm: boolean;
  disablePasswordReset: boolean;
  recaptcha: {
    enabled: boolean;
    siteKey: string;
    action: string;
  };
  session: {
    username: string | null;
  };
}

const LoginPage: FunctionComponent<Props> = ({
  authenticationProviders,
  disablePasswordReset,
  flashMessages,
  providerRequestPasswordResetPath,
  providerSessionsPath,
  recaptcha,
  session,
  show3scaleLoginForm
}) => (
  <PF4LoginPage
    backgroundImgAlt="Red Hat 3scale API Management"
    backgroundImgSrc={PF4DownstreamBG}
    brandImgAlt="Red Hat 3scale API Management"
    brandImgSrc={brandImg}
    forgotCredentials={show3scaleLoginForm && !disablePasswordReset && (
      <LoginMainFooterBandItem>
        <a
          href={providerRequestPasswordResetPath}
          // HACK: prevent click from missing link after input loses focus and component re-renders
          onMouseDown={(event) => { event.currentTarget.click() }}
        >
          Forgot password?
        </a>
      </LoginMainFooterBandItem>
    )}
    loginTitle="Log in to your account"
  >
    {show3scaleLoginForm && (
      <LoginForm
        flashMessages={flashMessages}
        providerSessionsPath={providerSessionsPath}
        recaptcha={recaptcha}
        session={session}
      />
    )}
    {authenticationProviders.length > 0 && (
      <div className="providers-separator">
        <AuthenticationProviders authenticationProviders={authenticationProviders} />
      </div>
    )}
  </PF4LoginPage>
)

// eslint-disable-next-line react/jsx-props-no-spreading
const LoginPageWrapper = (props: Props, containerId: string): void => { createReactWrapper(<LoginPage {...props} />, containerId) }

export type { Props }
export { LoginPage, LoginPageWrapper }
