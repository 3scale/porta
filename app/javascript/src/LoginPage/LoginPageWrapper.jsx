// @flow

import React from 'react'
import type { Node } from 'react'
import {createReactWrapper} from 'utilities/createReactWrapper'
import 'url-polyfill'

import {
  LoginPage
} from '@patternfly/react-core'

import {
  ForgotCredentials,
  Login3scaleForm,
  AuthenticationProviders,
  FlashMessages
} from 'LoginPage'

import 'LoginPage/assets/styles/loginPage.scss'

import brandImg from 'LoginPage/assets/images/3scale_Logo_Reverse.png'
import PF4DownstreamBG from 'LoginPage/assets/images/PF4DownstreamBG.svg'

import type { FlashMessage } from 'Types'
import type { ProvidersProps } from 'LoginPage'

type Props = {
  authenticationProviders: Array<ProvidersProps>,
  flashMessages: Array<FlashMessage>,
  providerAdminDashboardPath: string,
  providerLoginPath: string,
  providerSessionsPath: string,
  providerRequestPasswordResetPath: string,
  redirectUrl: string,
  show3scaleLoginForm: boolean,
  disablePasswordReset: boolean,
  session: {
    username: ?string
  }
}

type State = {
  loginTitle: string
}

class SimpleLoginPage extends React.Component<Props, State> {
  constructor (props: Props) {
    super(props)
    this.state = {
      loginTitle: 'Log in to your account'
    }
  }

  showForgotCredentials (): Node {
    const { disablePasswordReset, providerRequestPasswordResetPath, show3scaleLoginForm } = this.props
    const showResetPasswordLink = show3scaleLoginForm && !disablePasswordReset
    return showResetPasswordLink && <ForgotCredentials requestPasswordResetPath={providerRequestPasswordResetPath} />
  }

  loginForm (): Node {
    const hasAuthenticationProviders = this.props.authenticationProviders
    const show3scaleLoginForm = this.props.show3scaleLoginForm
    return (
      <React.Fragment>
        { show3scaleLoginForm &&
            <Login3scaleForm
              providerSessionsPath={this.props.providerSessionsPath}
              session={this.props.session}
            />
        }
        { hasAuthenticationProviders &&
          <div className='providers-separator'>
            <AuthenticationProviders
              authenticationProviders={this.props.authenticationProviders}
            />
          </div>
        }
      </React.Fragment>
    )
  }

  render (): Node {
    return (
      <LoginPage
        brandImgSrc={brandImg}
        brandImgAlt='Red Hat 3scale API Management'
        backgroundImgSrc={PF4DownstreamBG}
        backgroundImgAlt='Red Hat 3scale API Management'
        loginTitle={this.state.loginTitle}
        forgotCredentials={this.showForgotCredentials()}
        footer={null}
      >
        {
          this.props.flashMessages &&
          <FlashMessages flashMessages={this.props.flashMessages}/>
        }
        { this.loginForm() }
      </LoginPage>
    )
  }
}

const LoginPageWrapper = (props: Props, containerId: string): void =>
  createReactWrapper(<SimpleLoginPage {...props} />, containerId)

export {SimpleLoginPage, LoginPageWrapper}
