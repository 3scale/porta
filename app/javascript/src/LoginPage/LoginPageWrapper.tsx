import 'url-polyfill'
import { Component } from 'react'
import { createReactWrapper } from 'utilities/createReactWrapper'
import { LoginPage } from '@patternfly/react-core'
import { AuthenticationProviders } from 'LoginPage/loginForms/AuthenticationProviders'
import { FlashMessages } from 'LoginPage/loginForms/FlashMessages'
import { ForgotCredentials } from 'LoginPage/loginForms/ForgotCredentials'
import { Login3scaleForm } from 'LoginPage/loginForms/Login3scaleForm'
import brandImg from 'LoginPage/assets/images/3scale_Logo_Reverse.png'
import PF4DownstreamBG from 'LoginPage/assets/images/PF4DownstreamBG.svg'
import 'LoginPage/assets/styles/loginPage.scss'

import type { ProvidersProps } from 'LoginPage/loginForms/AuthenticationProviders'
import type { FlashMessage } from 'Types'

type Props = {
  authenticationProviders: Array<ProvidersProps>,
  flashMessages?: Array<FlashMessage>,
  providerAdminDashboardPath: string,
  providerLoginPath: string,
  providerSessionsPath: string,
  providerRequestPasswordResetPath: string,
  redirectUrl: string,
  show3scaleLoginForm: boolean,
  disablePasswordReset: boolean,
  session: {
    username: string | null | undefined
  }
}

type State = {
  loginTitle: string
}

class SimpleLoginPage extends Component<Props, State> {
  constructor (props: Props) {
    super(props)
    this.state = {
      loginTitle: 'Log in to your account'
    }
  }

  showForgotCredentials () {
    const { disablePasswordReset, providerRequestPasswordResetPath, show3scaleLoginForm } = this.props
    const showResetPasswordLink = show3scaleLoginForm && !disablePasswordReset
    return showResetPasswordLink && <ForgotCredentials requestPasswordResetPath={providerRequestPasswordResetPath} />
  }

  loginForm () {
    const hasAuthenticationProviders = this.props.authenticationProviders
    const show3scaleLoginForm = this.props.show3scaleLoginForm
    return (
      <>
        { show3scaleLoginForm && (
          <Login3scaleForm
            providerSessionsPath={this.props.providerSessionsPath}
            session={this.props.session}
          />
        )}
        { hasAuthenticationProviders && (
          <div className='providers-separator'>
            <AuthenticationProviders
              authenticationProviders={this.props.authenticationProviders}
            />
          </div>
        )}
      </>
    )
  }

  render () {
    return (
      <LoginPage
        backgroundImgAlt='Red Hat 3scale API Management'
        backgroundImgSrc={PF4DownstreamBG}
        brandImgAlt='Red Hat 3scale API Management'
        brandImgSrc={brandImg}
        forgotCredentials={this.showForgotCredentials()}
        loginTitle={this.state.loginTitle}
        // footer={null
      >
        { this.props.flashMessages && <FlashMessages flashMessages={this.props.flashMessages} /> }
        { this.loginForm() }
      </LoginPage>
    )
  }
}

// eslint-disable-next-line react/jsx-props-no-spreading
const LoginPageWrapper = (props: Props, containerId: string): void => createReactWrapper(<SimpleLoginPage {...props} />, containerId)

export { SimpleLoginPage, LoginPageWrapper, Props }
