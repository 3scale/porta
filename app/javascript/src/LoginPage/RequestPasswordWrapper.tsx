import { createReactWrapper } from 'utilities/createReactWrapper'
import { LoginPage } from '@patternfly/react-core'
import brandImg from 'LoginPage/assets/images/3scale_Logo_Reverse.png'
import PF4DownstreamBG from 'LoginPage/assets/images/PF4DownstreamBG.svg'
import 'LoginPage/assets/styles/loginPage.scss'
import { FlashMessages } from 'LoginPage/loginForms/FlashMessages'
import { RequestPasswordForm } from 'LoginPage/loginForms/RequestPasswordForm'

import type { FunctionComponent } from 'react'
import type { FlashMessage } from 'Types'

type Props = {
  flashMessages: Array<FlashMessage>,
  providerLoginPath: string,
  providerPasswordPath: string
}

const RequestPassword: FunctionComponent<Props> = ({
  flashMessages,
  providerLoginPath,
  providerPasswordPath
}) => (
  <LoginPage
    backgroundImgAlt='Red Hat 3scale API Management'
    backgroundImgSrc={PF4DownstreamBG}
    brandImgAlt='Red Hat 3scale API Management'
    brandImgSrc={brandImg}
    loginTitle='Request Password'
  >
    {flashMessages && <FlashMessages flashMessages={flashMessages} />}
    <RequestPasswordForm
      providerLoginPath={providerLoginPath}
      providerPasswordPath={providerPasswordPath}
    />
  </LoginPage>
)

// eslint-disable-next-line react/jsx-props-no-spreading
const RequestPasswordWrapper = (props: Props, containerId: string): void => createReactWrapper(<RequestPassword {...props} />, containerId)

export { RequestPassword, RequestPasswordWrapper, Props }
