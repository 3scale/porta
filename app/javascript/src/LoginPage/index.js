// @flow

export * from 'LoginPage/loginForms/ForgotCredentials'
export * from 'LoginPage/loginForms/Login3scaleForm'
export * from 'LoginPage/loginForms/RequestPasswordForm'
export * from 'LoginPage/loginForms/HiddenInputs'
export * from 'LoginPage/LoginPageWrapper'
export * from 'LoginPage/loginForms/AuthenticationProviders'
export * from 'LoginPage/loginForms/FlashMessages'
export * from 'LoginPage/loginForms/FormGroups'
export * from 'LoginPage/SignupPageWrapper'
export * from 'LoginPage/loginForms/SignupForm'
export * from 'LoginPage/utils/formValidation'
export * from 'LoginPage/RequestPasswordWrapper'
