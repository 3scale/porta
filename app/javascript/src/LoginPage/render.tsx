import { lazy, StrictMode, Suspense } from 'react'
import { render } from 'react-dom'

/* eslint-disable @typescript-eslint/no-unsafe-return */
// @ts-expect-error TS2691
const LoginPage = lazy(() => import('LoginPage/LoginPageWrapper.tsx'))
// @ts-expect-error TS2691
const RequestPassword = lazy(() => import('LoginPage/RequestPasswordWrapper.tsx'))
// @ts-expect-error TS2691
const SignupPage = lazy(() => import('LoginPage/SignupPageWrapper.tsx'))
// @ts-expect-error TS2691
const ChangePassword = lazy(() => import('ChangePassword/components/ChangePassword.tsx'))
/* eslint-enable @typescript-eslint/no-unsafe-return */

export default function (container: HTMLElement): void {
  const { loginProps, changePasswordProps, signupProps, requestProps } = container.dataset

  render((
    <StrictMode>
      <Suspense fallback="">
        {/* eslint-disable react/jsx-props-no-spreading */}
        {loginProps && <LoginPage {...JSON.parse(loginProps)} />}
        {changePasswordProps && <ChangePassword {...JSON.parse(changePasswordProps)} />}
        {signupProps && <SignupPage {...JSON.parse(signupProps)} />}
        {requestProps && <RequestPassword {...JSON.parse(requestProps)} />}
        {/* eslint-enable react/jsx-props-no-spreading */}
      </Suspense>
    </StrictMode>
  ), container)
}
