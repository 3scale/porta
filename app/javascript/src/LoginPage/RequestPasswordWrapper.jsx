// @flow

import * as React from 'react'
import { createReactWrapper } from 'utilities/createReactWrapper'

import { LoginPage } from '@patternfly/react-core'
import { RequestPasswordForm, FlashMessages } from 'LoginPage'

import 'LoginPage/assets/styles/loginPage.scss'

import brandImg from 'LoginPage/assets/images/3scale_Logo_Reverse.png'
import PF4DownstreamBG from 'LoginPage/assets/images/PF4DownstreamBG.svg'

import type { FlashMessage } from 'Types'

type Props = {
  flashMessages: Array<FlashMessage>,
  providerLoginPath: string,
  providerPasswordPath: string
}

const RequestPassword = ({ flashMessages, providerLoginPath, providerPasswordPath }: Props): React.Node => (
  <LoginPage
    brandImgSrc={brandImg}
    brandImgAlt='Red Hat 3scale API Management'
    backgroundImgSrc={PF4DownstreamBG}
    backgroundImgAlt='Red Hat 3scale API Management'
    loginTitle='Request Password'
  >
    {flashMessages && <FlashMessages flashMessages={flashMessages}/>}
    <RequestPasswordForm
      providerLoginPath={providerLoginPath}
      providerPasswordPath={providerPasswordPath}
    />
  </LoginPage>
)

const RequestPasswordWrapper = (props: Props, containerId: string): void =>
  createReactWrapper(<RequestPassword {...props} />, containerId)

export { RequestPassword, RequestPasswordWrapper }
