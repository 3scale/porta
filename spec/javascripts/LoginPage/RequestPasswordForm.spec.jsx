// @flow

import React from 'react'
import {mount} from 'enzyme'

import { RequestPasswordForm } from 'LoginPage'

const props = {
  flashMessages: [],
  providerLoginPath: 'login-path',
  providerPasswordPath: 'password-path'
}

it('should render itself', () => {
  const wrapper = mount(<RequestPasswordForm {...props}/>)
  expect(wrapper).toMatchSnapshot()
})

it('should render flashMessages when present', () => {
  const propsWithFlashMessages = {
    ...props,
    flashMessages: [{ message: 'Ooops', type: 'error' }]
  }
  const wrapper = mount(<RequestPasswordForm {...propsWithFlashMessages}/>)
  expect(wrapper).toMatchSnapshot()
})

it('should set email and validation state to true', () => {
  const event = {
    currentTarget: {
      value: 'bob@sponge.com',
      type: 'email'
    }
  }
  const wrapper = mount(<RequestPasswordForm {...props}/>)
  wrapper.find('input#email').props().onChange(event)
  expect(wrapper.state().email).toEqual('bob@sponge.com')
  expect(wrapper.state().validation.email).toEqual(true)
})

it('should set validation state to false when email is invalid', () => {
  const event = {
    currentTarget: {
      value: 'bobspongecom',
      type: 'email'
    }
  }
  const wrapper = mount(<RequestPasswordForm {...props}/>)
  wrapper.find('input#email').props().onChange(event)
  expect(wrapper.state().email).toEqual('bobspongecom')
  expect(wrapper.state().validation.email).toEqual(false)
})
