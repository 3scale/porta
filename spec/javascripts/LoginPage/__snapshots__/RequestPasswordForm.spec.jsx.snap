// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render flashMessages when present 1`] = `
<RequestPasswordForm
  flashMessages={
    Array [
      Object {
        "message": "Ooops",
        "type": "error",
      },
    ]
  }
  providerLoginPath="login-path"
  providerPasswordPath="password-path"
>
  <Form
    acceptCharset="UTF-8"
    action="password-path"
    id="request_password"
    method="post"
    noValidate={true}
  >
    <form
      acceptCharset="UTF-8"
      action="password-path"
      className="pf-c-form"
      id="request_password"
      method="post"
      noValidate={true}
    >
      <HiddenInputs
        isPasswordReset={true}
      >
        <input
          name="utf8"
          type="hidden"
          value="✓"
        />
        <input
          name="_method"
          type="hidden"
          value="delete"
        />
        <CSRFToken>
          <input
            type="hidden"
          />
        </CSRFToken>
      </HiddenInputs>
      <EmailField
        inputProps={
          Object {
            "autoFocus": "autoFocus",
            "fieldId": "email",
            "isRequired": true,
            "isValid": undefined,
            "label": "Email address",
            "name": "email",
            "onChange": [Function],
            "value": "",
          }
        }
      >
        <FormGroup
          helperTextInvalid="A valid email address is mandatory"
          inputProps={
            Object {
              "autoFocus": "autoFocus",
              "fieldId": "email",
              "isRequired": true,
              "isValid": undefined,
              "label": "Email address",
              "name": "email",
              "onChange": [Function],
              "value": "",
            }
          }
          type="email"
        >
          <FormGroup
            fieldId="email"
            helperTextInvalid="A valid email address is mandatory"
            isRequired={true}
            label="Email address"
          >
            <div
              className="pf-c-form__group"
            >
              <label
                className="pf-c-form__label"
                htmlFor="email"
              >
                <span
                  className="pf-c-form__label-text"
                >
                  Email address
                </span>
                <span
                  aria-hidden="true"
                  className="pf-c-form__label-required"
                >
                  *
                </span>
              </label>
              <ForwardRef
                autoFocus="autoFocus"
                id="email"
                isRequired={true}
                name="email"
                onChange={[Function]}
                type="email"
                value=""
              >
                <TextInputBase
                  aria-label={null}
                  autoFocus="autoFocus"
                  className=""
                  id="email"
                  innerRef={null}
                  isDisabled={false}
                  isReadOnly={false}
                  isRequired={true}
                  isValid={true}
                  name="email"
                  onChange={[Function]}
                  type="email"
                  validated="default"
                  value=""
                >
                  <input
                    aria-invalid={false}
                    aria-label={null}
                    autoFocus="autoFocus"
                    className="pf-c-form-control"
                    disabled={false}
                    id="email"
                    name="email"
                    onChange={[Function]}
                    readOnly={false}
                    required={true}
                    type="email"
                    value=""
                  />
                </TextInputBase>
              </ForwardRef>
            </div>
          </FormGroup>
        </FormGroup>
      </EmailField>
      <ActionGroup>
        <div
          className="pf-c-form__group pf-m-action"
        >
          <div
            className="pf-c-form__actions"
          >
            <Component
              className="pf-c-button pf-m-primary pf-m-block"
              isDisabled={true}
              type="submit"
            >
              <ComponentWithOuia
                component={[Function]}
                componentProps={
                  Object {
                    "children": "Reset password",
                    "className": "pf-c-button pf-m-primary pf-m-block",
                    "isDisabled": true,
                    "type": "submit",
                  }
                }
                consumerContext={null}
              >
                <Button
                  className="pf-c-button pf-m-primary pf-m-block"
                  isDisabled={true}
                  ouiaContext={
                    Object {
                      "isOuia": false,
                      "ouiaId": null,
                    }
                  }
                  type="submit"
                >
                  <button
                    aria-disabled={null}
                    aria-label={null}
                    className="pf-c-button pf-m-primary pf-c-button pf-m-primary pf-m-block"
                    disabled={true}
                    tabIndex={null}
                    type="submit"
                  >
                    Reset password
                  </button>
                </Button>
              </ComponentWithOuia>
            </Component>
            <a
              href="login-path"
            >
              Sign in
            </a>
          </div>
        </div>
      </ActionGroup>
    </form>
  </Form>
</RequestPasswordForm>
`;

exports[`should render itself 1`] = `
<RequestPasswordForm
  flashMessages={Array []}
  providerLoginPath="login-path"
  providerPasswordPath="password-path"
>
  <Form
    acceptCharset="UTF-8"
    action="password-path"
    id="request_password"
    method="post"
    noValidate={true}
  >
    <form
      acceptCharset="UTF-8"
      action="password-path"
      className="pf-c-form"
      id="request_password"
      method="post"
      noValidate={true}
    >
      <HiddenInputs
        isPasswordReset={true}
      >
        <input
          name="utf8"
          type="hidden"
          value="✓"
        />
        <input
          name="_method"
          type="hidden"
          value="delete"
        />
        <CSRFToken>
          <input
            type="hidden"
          />
        </CSRFToken>
      </HiddenInputs>
      <EmailField
        inputProps={
          Object {
            "autoFocus": "autoFocus",
            "fieldId": "email",
            "isRequired": true,
            "isValid": undefined,
            "label": "Email address",
            "name": "email",
            "onChange": [Function],
            "value": "",
          }
        }
      >
        <FormGroup
          helperTextInvalid="A valid email address is mandatory"
          inputProps={
            Object {
              "autoFocus": "autoFocus",
              "fieldId": "email",
              "isRequired": true,
              "isValid": undefined,
              "label": "Email address",
              "name": "email",
              "onChange": [Function],
              "value": "",
            }
          }
          type="email"
        >
          <FormGroup
            fieldId="email"
            helperTextInvalid="A valid email address is mandatory"
            isRequired={true}
            label="Email address"
          >
            <div
              className="pf-c-form__group"
            >
              <label
                className="pf-c-form__label"
                htmlFor="email"
              >
                <span
                  className="pf-c-form__label-text"
                >
                  Email address
                </span>
                <span
                  aria-hidden="true"
                  className="pf-c-form__label-required"
                >
                  *
                </span>
              </label>
              <ForwardRef
                autoFocus="autoFocus"
                id="email"
                isRequired={true}
                name="email"
                onChange={[Function]}
                type="email"
                value=""
              >
                <TextInputBase
                  aria-label={null}
                  autoFocus="autoFocus"
                  className=""
                  id="email"
                  innerRef={null}
                  isDisabled={false}
                  isReadOnly={false}
                  isRequired={true}
                  isValid={true}
                  name="email"
                  onChange={[Function]}
                  type="email"
                  validated="default"
                  value=""
                >
                  <input
                    aria-invalid={false}
                    aria-label={null}
                    autoFocus="autoFocus"
                    className="pf-c-form-control"
                    disabled={false}
                    id="email"
                    name="email"
                    onChange={[Function]}
                    readOnly={false}
                    required={true}
                    type="email"
                    value=""
                  />
                </TextInputBase>
              </ForwardRef>
            </div>
          </FormGroup>
        </FormGroup>
      </EmailField>
      <ActionGroup>
        <div
          className="pf-c-form__group pf-m-action"
        >
          <div
            className="pf-c-form__actions"
          >
            <Component
              className="pf-c-button pf-m-primary pf-m-block"
              isDisabled={true}
              type="submit"
            >
              <ComponentWithOuia
                component={[Function]}
                componentProps={
                  Object {
                    "children": "Reset password",
                    "className": "pf-c-button pf-m-primary pf-m-block",
                    "isDisabled": true,
                    "type": "submit",
                  }
                }
                consumerContext={null}
              >
                <Button
                  className="pf-c-button pf-m-primary pf-m-block"
                  isDisabled={true}
                  ouiaContext={
                    Object {
                      "isOuia": false,
                      "ouiaId": null,
                    }
                  }
                  type="submit"
                >
                  <button
                    aria-disabled={null}
                    aria-label={null}
                    className="pf-c-button pf-m-primary pf-c-button pf-m-primary pf-m-block"
                    disabled={true}
                    tabIndex={null}
                    type="submit"
                  >
                    Reset password
                  </button>
                </Button>
              </ComponentWithOuia>
            </Component>
            <a
              href="login-path"
            >
              Sign in
            </a>
          </div>
        </div>
      </ActionGroup>
    </form>
  </Form>
</RequestPasswordForm>
`;
